src/main/java/in/projecteka/gateway/user/UserController.java

```diff
@@ -41,4 +40,22 @@
         return patientSearchResponseOrchestrator.processResponse(requestEntity, X_HIU_ID)
                 .subscriberContext(context -> context.put("apiCalled", PATH_PATIENTS_ON_FIND));
     }
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @PostMapping(USERS_AUTH_CONFIRM)
+    public Mono<Void> authConfirm(HttpEntity<String> requestEntity){
+        return ReactiveSecurityContextHolder.getContext()
+                .map(securityContext -> (Caller) securityContext.getAuthentication().getPrincipal())
+                .map(Caller::getClientId)
+                .flatMap(clientId -> authConfirmRequestOrchestrator
+                        .handleThis(requestEntity, X_CM_ID, X_HIP_ID, clientId)
+                        .subscriberContext(context -> context.put(API_CALLED, USERS_AUTH_CONFIRM)));
+    }
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @PostMapping(USERS_AUTH_ON_CONFIRM)
+    public Mono<Void> authOnConfirm(HttpEntity<String> requestEntity){
+        return Mono.empty();
```

Comment:
will there be implementation coming?

Reply:
yeah from here apporva story kicks in

Reply:
@Nexengineer We are working on `/links/link/add-contexts` and `/links/link/on-add-contexts` apis on gateway. I don't think it has anything to do with onConfirm.

Reply:
On this "on-confirm" you will get the token. Didn't knew where to keep it so stopped here

Reply:
We had discussed this during kick off right. Saying you will have to implement(Log request)on-confirm on hip  and gateway should call that. That way we can test the flow as well.

Reply:
done